### PR TITLE
don't use an exclusive transaction on _fishbowl collection (#13491)

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/13491

Don't use an exclusive transaction on _fishbowl collection every time the Foxx store results are written to that collection. 
Instead perform two separate operations: insert-replace and remove (the latter only if necessary).

The overall operation will not be atomic though, but it is more efficient than a truncate and then re-inserting everything. Moving away from exclusive locks is good for other reasons, too.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13870/

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
+  - `arangodb_document_remove_time`: Execution time histogram of all
+    document remove operations (successful and failed) [s].
+  - `arangodb_document_update_time`: Execution time histogram of all
+    document update operations (successful and failed) [s].
   - `arangodb_collection_truncate_time`: Execution time histogram of all
     collection truncate operations (successful and failed) [s].
 
+  The timer metrics are turned off by default, and can be enabled by setting
+  the startup option `--server.export-read-write-metrics true`.
+
```
Comment (mchacki) [inline CHANGELOG:38]: This piece looks a bit unrelated.
Is it possible that this is a hidden merge conflict?

**Review by mchacki — CHANGES_REQUESTED** [review]
The code itself is good and ready to go.
But I think there is some unintended
commit inside the CHANGELOG.
Not need to rerun Jenkins if the comment is fixed.

File: CHANGELOG (hunk 1 above)
Comment (jsteemann) [reply CHANGELOG:38]: The CHANGELOG change is undeed unrelated to this PR, but I put it in as a clarification.
The clarification was added to devel already, but was missing in 3.7.
So it is intentional as it is.
Of course I could have split this into 2 separate PRs, just didn't due to lack of :hourglass:

Comment (mchacki) [reply CHANGELOG:38]: ok 👍

**Review by mchacki — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
Restarted Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13875/

**Comment by jsteemann** [comment]
Tests blue

**Review by KVS85 — APPROVED** [review]
LGTM.
